### ISSUE TITLE
[REVIEW]: tabulog: A Language-Agnostic Template System for Parsing Log Files

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1917]
**Submitting author:** @austinnar (<a href="http://orcid.org/0000-0002-5319-9805">Austin Nar</a>)
**Repository:** <a href="https://github.com/AustinNar/tabulog" target ="_blank">https://github.com/AustinNar/tabulog</a>
**Version:** 0.1.1
**Editor:** @trallard
**Reviewer:** @alexanderfurnas, @trallard
**Archive:** Pending

## Status

[![status](https://joss.theoj.org/papers/4c1713bdabfca57b582a4b26ac9f8ba2/status.svg)](https://joss.theoj.org/papers/4c1713bdabfca57b582a4b26ac9f8ba2)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/4c1713bdabfca57b582a4b26ac9f8ba2"><img src="https://joss.theoj.org/papers/4c1713bdabfca57b582a4b26ac9f8ba2/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/4c1713bdabfca57b582a4b26ac9f8ba2/status.svg)](https://joss.theoj.org/papers/4c1713bdabfca57b582a4b26ac9f8ba2)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/AustinNar/tabulog" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@alexanderfurnas & @trallard, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @trallard know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/AustinNar/tabulog">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@austinnar) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/AustinNar/tabulog">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@austinnar) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @alexanderfurnas, @trallard it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by trallard** [comment]
Hey @alexanderfurnas I have started the review. Your checklist can be found at the top of this issue.

If you encounter any problems feel free to ping me

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01917/joss.01917/10.21105.joss.01917.pdf)

**Comment by trallard** [comment]
@whedon remind @alexanderfurnas in 1 week

**Comment by whedon** [comment]
Reminder set for @alexanderfurnas in 1 week

**Comment by trallard** [comment]
@whedon remind @trallard in 1 week

**Comment by whedon** [comment]
Reminder set for @trallard in 1 week

**Comment by trallard** [comment]
@AustinNar I have created a couple of issues on the repo that will need to be dealt with for the review to move forward

**Comment by whedon** [comment]
:wave: @alexanderfurnas, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @trallard, please update us on how your review is going.

**Comment by trallard** [comment]
This review is currently blocked by actions that need to be taken on the author side, @AustinNar, can you update us on this please?

**Comment by trallard** [comment]
@AustinNar have you got any updates on this?
@alexanderfurnas  have you got any time to help with the review?

**Comment by danielskatz** [comment]
👋 @AustinNar - please let us know the status of your updates

**Comment by danielskatz** [comment]
👋 @AustinNar - without a response soon, we will be forced to reject this submission

**Comment by AustinNar** [comment]
Hi, I am so sorry, these emails were getting routed to my junk folder. I
will take these actions this week.

**Comment by danielskatz** [comment]
great!

**Comment by AustinNar** [comment]
Hello! Sorry for the long delay before actually beginning work on this.

I have added a CONTRIBUTING.md which is linked from main README. I added
installation instructions and links to R and Python examples both in the
main README and in a (new) docs/README.md file. Let me know if you would
like any of this to be more comprehensive.

Thanks!

**Comment by danielskatz** [comment]
👋 @trallard - this seems like it is waiting for you to take action, either as a review or as the editor

**Comment by trallard** [comment]
As an update: I will pick up this back on Monday

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by arfon** [comment]
:wave: @trallard &@alexanderfurnas, just a friendly check-in to see how things are going with your reviews?

**Comment by arfon** [comment]
👋 @trallard & @alexanderfurnas, today we [reopened JOSS](https://blog.joss.theoj.org/2020/05/reopening-joss) for new submissions and are checking in on our existing reviews. Do you think you might be able to wrap up your reviews in the next 2-3 weeks?

**Comment by trallard** [comment]
Sure, I have blocked time on Monday to complete this. 

@alexanderfurnas  have you got any updates?

**Comment by kthyng** [comment]
Hi! It looks like we need more on the reviews from both @alexanderfurnas and @trallard. I will also email @alexanderfurnas now since he hasn't participated on this review issue at all.

**Comment by kthyng** [comment]
Heard back from @alexanderfurnas and is ok for the review in the next two weeks.

**Comment by labarba** [comment]
hi 👋  — it looks like this review fell in the back burner again?  @alexanderfurnas and @trallard : can you give us a status update?

**Comment by trallard** [comment]
@alexanderfurnas I am planning to complete my review today/tomorrow is there any chance you can update us on your status?

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by trallard** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.01917/joss.01917/10.21105.joss.01917.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.01917/joss.01917/10.21105.joss.01917.pdf) :page_facing_up: :point_left:

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.01917/joss.01917/10.21105.joss.01917.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.01917/joss.01917/10.21105.joss.01917.pdf) :page_facing_up: :point_left:

**Comment by trallard** [comment]
@AustinNar I have opened a couple of issues in the repo 

Also I have emailed Alexander to check the status of his review. Shall he not be able to complete it I can find another reviewer to expedite this submission

**Comment by arfon** [comment]
@trallard - happy new year! I think we should go ahead and find an alternative reviewer here.

**Comment by arfon** [comment]
:wave: @AustinNar - as part of a review I've been doing this week as EiC of JOSS of the outstanding submissions I've spent some time reviewing this submission. Unfortunately I believe this submission should have been identified as out of scope during the [pre-review](https://github.com/openjournals/joss-reviews/issues/1680) stage of the process. In particular, this submission is out of scope due to it's small size and [limited scholarly effort](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort).

I realize this submission has been under review for quite some time and I apologize that this was not flagged earlier in the process before @trallard and @alexanderfurnas started their reviews here.

**Comment by arfon** [comment]
@whedon reject

**Comment by whedon** [comment]
Paper rejected.
